diff/diff.go

```diff
@@ -0,0 +1,534 @@
+	case "delete":
+		kvs := make([]string, 0, len(keys))
+		for _, col := range keys {
+			if needQuotes(col.FieldType) {
+				kvs = append(kvs, fmt.Sprintf("%s = \"%s\"", col.Name.O, string(data[col.Name.O])))
+			} else {
+				kvs = append(kvs, fmt.Sprintf("%s = %s", col.Name.O, string(data[col.Name.O])))
+			}
+		}
+		sql = fmt.Sprintf("DELETE FROM `%s`.`%s` where %s;", dbName, table.Name, strings.Join(kvs, " AND "))
+	default:
+		log.Errorf("unknow sql type %s", tp)
+	}
+
+	return
+}
+
+func needQuotes(ft types.FieldType) bool {
+	return !(pkgdb.IsNumberType(ft.Tp) || pkgdb.IsFloatType(ft.Tp))
+}
+
+func compareData(map1 map[string][]byte, map2 map[string][]byte, orderKeyCols []*model.ColumnInfo) (bool, int32, error) {
+	var (
+		equal        = true
+		data1, data2 []byte
+		key          string
+		ok           bool
+		cmp          int32
+	)
+
+	for key, data1 = range map1 {
+		if data2, ok = map2[key]; !ok {
+			return false, 0, errors.Errorf("don't have key %s", key)
+		}
+		if string(data1) == string(data2) {
+			continue
+		}
+		equal = false
+		log.Errorf("find difference data, data1: %s, data2: %s", map1, map2)
+		break
+	}
+	if equal {
+		return true, 0, nil
+	}
+
+	for _, col := range orderKeyCols {
+		if data1, ok = map1[col.Name.O]; !ok {
+			return false, 0, errors.Errorf("don't have key %s", col.Name.O)
+		}
+		if data2, ok = map2[col.Name.O]; !ok {
+			return false, 0, errors.Errorf("don't have key %s", col.Name.O)
+		}
+		if needQuotes(col.FieldType) {
+			if string(data1) > string(data2) {
+				cmp = 1
+				break
+			} else if string(data1) < string(data2) {
+				cmp = -1
+				break
+			} else {
+				continue
+			}
+		} else {
+			num1, err1 := strconv.ParseFloat(string(data1), 64)
+			num2, err2 := strconv.ParseFloat(string(data2), 64)
+			if err1 != nil || err2 != nil {
+				return false, 0, errors.Errorf("convert %s, %s to float failed, err1: %v, err2: %v", string(data1), string(data2), err1, err2)
+			}
+			if num1 > num2 {
+				cmp = 1
+				break
+			} else if num1 < num2 {
+				cmp = -1
+				break
+			} else {
+				continue
+			}
+		}
+	}
+
+	return false, cmp, nil
+}
+
+func getChunkRows(db *sql.DB, dbName string, table *TableCheckCfg, where string, useRowID bool) (*sql.Rows, []*model.ColumnInfo, error) {
+	orderKeys, orderKeyCols := pkgdb.GetOrderKey(table.Info)
+	columns := "*"
+	if orderKeys[0] == pkgdb.ImplicitColName {
+		columns = fmt.Sprintf("*, %s", pkgdb.ImplicitColName)
+	}
+	query := fmt.Sprintf("SELECT /*!40001 SQL_NO_CACHE */ %s FROM `%s`.`%s` WHERE %s ORDER BY %s",
+		columns, dbName, table.Name, where, strings.Join(orderKeys, ","))
+
+	rows, err := pkgdb.QuerySQL(db, query)
+	if err != nil {
+		return nil, nil, errors.Trace(err)
+	}
+	return rows, orderKeyCols, nil
+}
+
+func equalStrings(str1, str2 []string) bool {
+	if len(str1) != len(str2) {
+		return false
+	}
+	for i := 0; i < len(str1); i++ {
```

Comment:
too slow

Reply:
it is just used to judge table name equal, and how to refine it

Reply:
if u only use it to check tables, no matter